src/overlays/actors/ovl_En_Ruppecrow/z_en_ruppecrow.c

```diff
@@ -81,62 +112,569 @@ static DamageTable D_80BE39EC = {
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ruppecrow/func_80BE2D4C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ruppecrow/func_80BE2E18.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ruppecrow/func_80BE2F6C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ruppecrow/func_80BE30F4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ruppecrow/func_80BE3178.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ruppecrow/func_80BE32DC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ruppecrow/func_80BE3354.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ruppecrow/func_80BE348C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ruppecrow/func_80BE35A4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ruppecrow/EnRuppecrow_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ruppecrow/EnRuppecrow_Destroy.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ruppecrow/EnRuppecrow_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ruppecrow/EnRuppecrow_Draw.s")
+s32 EnRuppecrow_UpdateCollision(EnRuppecrow* this, GlobalContext* globalCtx) {
+    UNK_TYPE pad;
+
+    this->collider.elements->dim.worldSphere.center.x = this->actor.world.pos.x;
+    this->collider.elements->dim.worldSphere.center.y =
+        sJntSphInit.elements->dim.modelSphere.center.y + this->actor.world.pos.y;
+    this->collider.elements->dim.worldSphere.center.z = this->actor.world.pos.z;
+
+    CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+    Actor_UpdateBgCheckInfo(globalCtx, &this->actor, 12.0f, 25.0f, 50.0f, 0x07);
+
+    return true;
+}
+
+s32 EnRuppecrow_ReachedNextPoint_Clockwise(EnRuppecrow* this, Path* path, s32 index) {
+    Vec3s* points;
+    s32 pathCount;
+    s32 pathIndex;
+    s32 result;
+    f32 y;
+    f32 x;
+    Vec3f vec;
+    Vec3f point;
+
+    pathIndex = index;
+    points = Lib_SegmentedToVirtual(path->points);
+    pathCount = path->count;
+    result = false;
+
+    Math_Vec3s_ToVec3f(&point, &points[pathIndex]);
+
+    if (pathIndex == 0) {
+        y = points[1].x - points[0].x;
+        x = points[1].z - points[0].z;
+    } else {
+        if (pathIndex == (pathCount - 1)) {
+            y = points[pathCount - 1].x - points[pathCount - 2].x;
+            x = points[pathCount - 1].z - points[pathCount - 2].z;
+        } else {
+            y = points[pathIndex + 1].x - points[pathIndex - 1].x;
+            x = points[pathIndex + 1].z - points[pathIndex - 1].z;
+        }
+    }
+
+    func_8017B7F8(&point, (func_80086B30(y, x) * 10430.378f), &vec.z, &vec.y, &vec.x);
+    if (((this->actor.world.pos.x * vec.z) + (vec.y * this->actor.world.pos.z) + vec.x) > 0.0f) {
+        result = true;
+    }
+
+    return result;
+}
+
+s32 EnRuppecrow_ReachedNextPoint_CounterClockwise(EnRuppecrow* this, Path* path, s32 index) {
```

Comment:
I'd probably also remove the Next from these functions as well, since you are passing the interested pointIndex in as an argument.